File: 백준/Gold/2195. 문자열 복사/문자열 복사.py
def solve(S, P):
    n, m = len(S), len(P)
    i = 0
    count = 0
    
    while i < m:
        # 찾을 수 있는 최대 길이의 부분 문자열을 찾는다.
        max_len = 0
        for j in range(n):
            if S[j] == P[i]:
                length = 0
                while j + length < n and i + length < m and S[j + length] == P[i + length]:
                    length += 1
                max_len = max(max_len, length)
        
        # 찾은 최대 길이만큼 이동
        i += max_len
        count += 1
    
    return count

# 입력 받기
S = input().strip()
P = input().strip()

# 결과 출력
print(solve(S, P))

File: 백준/Gold/2195. 문자열 복사/문자열 복사.js
const stepOfcopy = (idx, S, P) =>{
    let rtn = 0;
    let j = 0
    for(let i=0; i < S.length; i++){
        if (S[i] === P[idx]){
            j = 0;
            while (i+j < S.length && idx+j < P.length && S[i+j] === P[idx+j]){
                j++;
            }
            rtn = Math.max(rtn, j)
        }
    }
    return rtn;
}


const solve = (S, P) => {
    let rtn = 0;
    let i = 0
    while (i < P.length){
        i += stepOfcopy(i, S, P);
        rtn += 1
    }
    return rtn;
}

const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];
rl.on('line', line => {
    input.push(line);
	if (input.length === 2) rl.close();
}).on('close', () => {
    const S = [...input[0]];
    const P = [...input[1]];
    console.log(solve(S, P))
});
